src/pages/home/report/ReportActionItemGrouped.js

```diff
@@ -4,16 +4,35 @@ import PropTypes from 'prop-types';
 import ReportActionPropTypes from './ReportActionPropTypes';
 import ReportActionItemMessage from './ReportActionItemMessage';
 import styles from '../../../styles/styles';
+import ReportActionItemIOUPreview from '../../../components/ReportActionItemIOUPreview';
 
 const propTypes = {
     // All the data of the action
     action: PropTypes.shape(ReportActionPropTypes).isRequired,
+
+    // Is this the most recent IOU Action?
+    isMostRecentIOUReport: PropTypes.bool.isRequired,
+
+    // The report currently being looked at
+    report: PropTypes.shape({
+
+        // IOU report ID associated with current report
+        iouReportID: PropTypes.number,
+    }).isRequired,
 };
 
-const ReportActionItemGrouped = ({action}) => (
+const ReportActionItemGrouped = ({action, report, isMostRecentIOUReport}) => (
     <View style={[styles.chatItem]}>
         <View style={[styles.chatItemRightGrouped]}>
-            <ReportActionItemMessage action={action} />
+            {action.actionName === 'IOU'
```

Comment:
NAB, I think at this point this component has probably outgrown it's original design and should be refactored eventually.

e.g. 

We have a group view and ungrouped view and instead of adding two checks in both components to figure out what content to render we probably should be passing in the message content as a child from a parent. This way we do not need to have duplicate logic in two components as if anything about them changes in the future we must remember to update both the grouped and ungrouped case.

Reply:
I refactored this part.